src/stream_base.cc

```diff
@@ -414,7 +412,11 @@ void StreamBase::EmitData(ssize_t nread,
   if (argv[2].IsEmpty())
     argv[2] = Undefined(env->isolate());
 
-  GetAsyncWrap()->MakeCallback(env->onread_string(), ARRAY_SIZE(argv), argv);
+  node::MakeCallback(env,
+                     GetObject(),
+                     env->onread_string(),
+                     ARRAY_SIZE(argv),
+                     argv);
```

Comment:
no. it gets around some logic that allows better performance than `node::MakeCallback()`.